dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageAPIImpl.java

```diff
@@ -64,92 +83,102 @@ public boolean isAssetTypeLanguage(String id) {
 		return factory.getLanguage(id);
-
 	}
 
+	@Override
 	public List<Language> getLanguages() {
 		return factory.getLanguages();
-
 	}
 
-	public void saveLanguage(Language o) {
-		factory.saveLanguage(o);
+	@Override
+	public void saveLanguage(final Language language) {
+		factory.saveLanguage(language);
 	}
 
-	public String getLanguageCodeAndCountry(long id, String langId) {
+	@Override
+	public String getLanguageCodeAndCountry(final long id, final String langId) {
 		return factory.getLanguageCodeAndCountry(id, langId);
-
 	}
 
+	@Override
 	public Language getDefaultLanguage() {
 		return factory.getDefaultLanguage();
-
 	}
 
-	public boolean hasLanguage(String id) {
+	@Override
+	public boolean hasLanguage(final String id) {
 		return factory.hasLanguage(id);
 	}
 
-	public boolean hasLanguage(long id) {
+	@Override
+	public boolean hasLanguage(final long id) {
 		return factory.hasLanguage(id);
 	}
 
-	public boolean hasLanguage(String languageCode, String countryCode) {
+	@Override
+	public boolean hasLanguage(final String languageCode, final String countryCode) {
 		return factory.hasLanguage(languageCode, countryCode);
 	}
 
-	public List<LanguageKey> getLanguageKeys(String langCode) {
-		List<LanguageKey> list = factory.getLanguageKeys(langCode);
-		Collections.sort(list, new LanguageKeyComparator());
+	@Override
+	public List<LanguageKey> getLanguageKeys(final String langCode) {
+		final List<LanguageKey> list = factory.getLanguageKeys(langCode);
+		Collections.sort(list, LANGUAGE_KEY_COMPARATOR);
 		return list;
 	}
 
-	public List<LanguageKey> getLanguageKeys(String langCode, String countryCode) {
-		List<LanguageKey> list = factory.getLanguageKeys(langCode, countryCode);
-		Collections.sort(list, new LanguageKeyComparator());
+	@Override
+	public List<LanguageKey> getLanguageKeys(final String langCode, final String countryCode) {
+		final List<LanguageKey> list = factory.getLanguageKeys(langCode, countryCode);
+		Collections.sort(list, LANGUAGE_KEY_COMPARATOR);
 		return list;
 	}
 
-	public List<LanguageKey> getLanguageKeys(Language lang) {
-		String langCode = lang.getLanguageCode();
-		String countryCode = lang.getCountryCode();
-		List<LanguageKey> list = new ArrayList<LanguageKey>();
-		list.addAll(factory.getLanguageKeys(langCode));
-		Collections.sort(list, new LanguageKeyComparator());
+	@Override
+	public List<LanguageKey> getLanguageKeys(final Language lang) {
+		final String langCode = lang.getLanguageCode();
+        final String countryCode = lang.getCountryCode();
+		final List<LanguageKey> list = new ArrayList<LanguageKey>(factory.getLanguageKeys(langCode));
+		Collections.sort(list, LANGUAGE_KEY_COMPARATOR);
 
-		List<LanguageKey> keys = factory.getLanguageKeys(langCode, countryCode);
-		for(LanguageKey key : keys) {
+		final List<LanguageKey> keys = factory.getLanguageKeys(langCode, countryCode);
+		for(LanguageKey key : keys) { // todo: analize it but it could be used an set instead of arraylist.
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)